doc/guides/modules.rst

```diff
@@ -10,14 +10,342 @@ system those are called *modules*. These modules must be integrated with the
 Zephyr build system, as described in more detail in other sections on
 this page.
 
-Zephyr depends on several categories of modules, including:
+To be classified as a candidate for being included in the default list of
+modules, an external project is required to have its own life-cycle outside
+the Zephyr Project, that is, reside in its own repository, and have its own
+contribution and maintainance workfow and release process. Zephyr modules
+should not contain code that is written exclusively for Zephyr.
+
+  .. note::
+
+     Code which is written for Zephyr and works exclusively with Zephyr
+     should not be submitted as a Zephyr module; instead, it should be
+     part of the main Zephyr tree.
+
+Modules to be included in the default manifest of the Zephyr project need to
+provide functionality or features endorsed and approved by the project technical
+steering committee and should comply with the
+:ref:`module licensing requirements<modules_licensing>` and
+:ref:`contribute guidelines<modules_contributing>`. They should also have a
+Zephyr developer that is commited to maintain the module code base.
+
+Zephyr depends on several categories of modules, including but not limited to:
 
 - Debugger integration
 - Silicon vendor Hardware Abstraction Layers (HALs)
 - Cryptography libraries
 - File Systems
 - Inter-Process Communication (IPC) libraries
 
+This page summarizes a list of policies and best practices which aim at
+better organizing the workflow in Zephyr modules.
+
+Module Repositories
+*******************
+
+* All modules included in the default manifest shall be hosted in repositories
+  under the Zephyr Project Github organization.
+
+* Module repository names should follow the convention of using lowercase
+  letters and dashes instead of underscores. This rule will apply to all
+  new module repositories.
+
+  .. note::
+
+     It is not required to modify the names of existing module repositories
+     to make them comply with the above naming convention.
+
+* If the module has an external (upstream) project repository, the module
+  repository should preserve the upstream repository folder structure.
+
+* Modules should use "zephyr" as the default name for the repository main
+  branch. Branches for specific purposes, for example, a module branch for
+  an LTS Zephyr version, shall have names starting with the 'zephyr\_' prefix.
+
+  .. note::
+
+     It is **not** recommended in module repositories to have a 'master'
+     branch mirroring the master branch of the external repository.
```

Comment:
Thanks for the comment @daor-oti @pabigot 
This was discussed in the meeting.
The main reason for not having a mirror for upstream main branch (most likely "master") is to make sure there won't be any accidents with manifest SHAs pointing to a module's "master" branch instead of "zephyr".
It was discussed at the meeting that there is no particular advantage for a Maintainer in a module, to keep tracking the module's upstream master branch. It could only cause confusion downstream. 

We have also agreed that zephyr modules should not, really, maintain other branches than "zephyr", which is to be used in zephyr manifest.

What _is_ important for maintainers in modules is to add a second "remote" to be able to synchronize fro upstream, but they do not need to keep the upstream main branch downstream.

Reply:
I will add a sentence for the rationale for not maintaining the upstream branch in the downstream repo.